File: backend/src/main/java/com/feedback/entity/Category.java
package com.feedback.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 反馈类别实体
 *
 * @author Claude
 * @since 2026-03-11
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("tb_category")
public class Category extends BaseEntity {

    private static final long serialVersionUID = 1L;

    /**
     * 类别名称
     */
    private String name;

    /**
     * 类别描述
     */
    private String description;

    /**
     * 是否关联任课教师：1=是，0=否
     */
    private Integer requireTeacher;

    /**
     * 排序（数字越小越靠前）
     */
    private Integer sort;

    /**
     * 状态：1=启用，0=禁用
     */
    private Integer status;

}

File: frontend/src/types/category.ts
/**
 * 反馈类别相关类型定义
 */

// 反馈类别
export interface Category {
  id: number
  name: string
  description?: string
  requireTeacher: boolean // 是否必须选择教师
  icon?: string
  sort: number // 排序
  createTime: string
}

// 创建类别请求
export interface CreateCategoryRequest {
  name: string
  description?: string
  requireTeacher: boolean
  icon?: string
  sort?: number
}

// 更新类别请求
export interface UpdateCategoryRequest {
  id: number
  name?: string
  description?: string
  requireTeacher?: boolean
  icon?: string
  sort?: number
}
